cloudify_rest_client/secrets.py

```diff
@@ -89,15 +93,26 @@ def create(self,
         :param visibility: The visibility of the secret, can be 'private',
                            'tenant' or 'global'
         :type visibility: unicode
+        :param schema: A JSON schema against which the secret will be validated
+        :type schema: dict
         :returns: New secret metadata
         :rtype: Dict[str]
 
         """
+        if schema and schema.get('type') != 'string':
+            try:
+                value = json.loads(value)
+            except json.decoder.JSONDecodeError as e:
+                raise CloudifyClientError(
+                    f'Error decoding JSON string "{value}" as '
+                    f'type "{schema.get("type")}": {e.args[0]}')
```

Comment:
maybe no need for this validation. Server-side is enough? Especially as it's not correct :P
eg. `.create(value=1, schema={'type': 'number'})`

Reply:
We need to handle `json.loads(value)` failures gracefully somehow. I can add a TypeError here